File: creational/builder/typescript/product.ts
export class Pizza {
	public size: string | null = null;
	public toppings: string[] | null = null;
	public crustType: string | null = null;
	public sauce: string | null = null;
	public cheese: string | null = null;

	describe(): string {
		const description: string[] = [];
		description.push(`This is a ${this.size} pizza.`);

		for (const [name, value] of Object.entries(this)) {
			if (value && name !== 'size') {
				let formattedName = name[0].toUpperCase() + name.substring(1);
				let formattedValue: string;
				if (Array.isArray(value)) {
					formattedValue = value.join(', ');
				} else {
					formattedValue = value.toString();
				}
				description.push(`${formattedName}: ${formattedValue}`);
			}
		}

		return description.join('\n');
	}
}

File: creational/builder/python/product.py
class Pizza:
    def __init__(
        self,
        size: str = None,
        toppings: list[str] = None,
        crust_type: str = None,
        sauce: str = None,
        cheese: str = None,
    ):
        self.size: str | None = size
        self.toppings: list[str] | None = toppings
        self.crust_type: str | None = crust_type
        self.sauce: str | None = sauce
        self.cheese: str | None = cheese

    def __str__(self):
        description = [
            f"This is a {self.size} pizza.",
            *[
                f"{name.capitalize()}:"
                f" {', '.join(value) if isinstance(value, list) else value}"
                for name, value in self.__dict__.items()
                if value is not None and name != "_size"
            ],
        ]
        return "\n".join(description)
